wdl/Module00c.wdl

```diff
@@ -288,7 +289,7 @@ workflow Module00c {
   call bem.EvidenceMerging as EvidenceMerging {
     input:
       samples = all_samples,
-      BAF_files = if defined(BAFFromShardedVCF.baf_files) then BAFFromShardedVCF.baf_files else BAF_files,
+      BAF_files = if defined(BAFFromShardedVCF.baf_files) then BAFFromShardedVCF.baf_files else (if defined(BAF_files) then select_all(select_first([BAF_files])) else NONE_ARRAY_FILE_),
```

Comment:
- BAFFromShardedVCF.baf_files -> Array[File]?
- BAF_files -> Array[File?]?

Both of which are valid, but the spec disallows this:

>  The return type of the if-then-else should be the same, regardless of which side is evaluated or runtime problems might occur. [[source](https://github.com/openwdl/wdl/blob/main/versions/1.0/SPEC.md#:~:text=%20the%20return%20type%20of%20the%20if-then-else%20should%20be%20the%20same%2C%20regardless%20of%20which%20side%20is%20evaluated%20or%20runtime%20problems%20might%20occur.)]

Error from MiniWDL:

> Expected Array[File]? instead of Array[File?]?  (if consequent & alternative must have the same type)

Reply:
> (if consequent & alternative must have the same type)

That error message is from a relatively old version of miniwdl. I think upgrading would resolve this case where it wasn't "unifying" the optionality of the array item type. (which is different from https://github.com/openwdl/wdl/issues/462, where I think you found a corner of the spec that isn't totally clear)

Reply:
😓  https://github.com/broadinstitute/wdl-ide/blob/76a04051677347fb9014ff6ef9c491b197f89d4a/server/setup.py#L34

```
wdl-lsp 0.0.72 requires miniwdl==0.5.1
```

---

Can confirm, miniwdl==1.1.5 validates the original correctly, I've reverted this portion.